scipy/io/netcdf.py

```diff
@@ -987,6 +983,50 @@ def __setitem__(self, index, data):
                 self.data.resize(shape)
         self.data[index] = data
 
+    def _get_missing_value(self):
+        """
+        Returns the value denoting "no data" for this variable.
+
+        If this variable does not have a missing/fill value, returns None.
+        """
+
+        if ('_FillValue' in self._attributes):
+            missing_value = self._attributes['_FillValue']
+        elif ('missing_value' in self._attributes):
+            missing_value = self._attributes['missing_value']
+        else:
+            missing_value = None
+
+        return missing_value
+
+    @staticmethod
+    def _apply_missing_value(data, missing_value):
+        """
+        Applies the given missing value to the data array.
+
+        Returns a numpy.ma array, with any value equal to missing_value masked
+        out (unless missing_value is None, in which case the original array is
+        returned).
+        """
+
+        if missing_value is None:
+            newdata = data
+        else:
+            try:
+                missing_value_isnan = np.isnan(missing_value)
+            except (TypeError, NotImplementedError):
+                # some data types (e.g., characters) cannot be tested for NaN
+                missing_value_isnan = False
+
+            if (missing_value_isnan):
+                mymask = np.isnan(data)
```

Comment:
> This doesn't look right. You should only use isnan if missing_value is actually np.nan right?

Sorry, I don't understand your suggestion. Note that ```mymask = np.isnan(data)``` is already in a conditional that checks ```missing_value_isnan``` (which is set above). Is there something else you're suggesting?

>  What if you have an integer input? np.isnan(intarray) will work fine, but intarray cannot contain any nan values.

There is the possibility that a dataset will provide a missing value of NaN on integer data. To me, though, that's a mistake in the dataset (the type of the ```_FillValue``` attribute should agree with the type of the data), and I'm having trouble seeing any handling we can / should do of cases like this.